library/src/main/java/com/novoda/downloadmanager/DownloadMigrator.java

```diff
@@ -0,0 +1,60 @@
+package com.novoda.downloadmanager;
+
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.ServiceConnection;
+import android.os.Handler;
+import android.os.IBinder;
+
+public class DownloadMigrator {
+
+    private final Context applicationContext;
+    private final Handler handler;
+    private final NotificationChannelCreator notificationChannelCreator;
+    private final NotificationCreator<MigrationStatus> notificationCreator;
+
+    DownloadMigrator(Context context,
+                     Handler handler,
+                     NotificationChannelCreator notificationChannelCreator,
+                     NotificationCreator<MigrationStatus> notificationCreator) {
+        this.applicationContext = context.getApplicationContext();
+        this.handler = handler;
+        this.notificationChannelCreator = notificationChannelCreator;
+        this.notificationCreator = notificationCreator;
+    }
+
+    public void startMigration(final MigrationCallback migrationCallback) {
+        ServiceConnection serviceConnection = new ServiceConnection() {
+            @Override
+            public void onServiceConnected(ComponentName name, IBinder binder) {
+                DownloadMigrationService migrationService = ((LiteDownloadMigrationService.MigrationDownloadServiceBinder) binder).getService();
+                migrationService.setNotificationChannelCreator(notificationChannelCreator);
+                migrationService.setNotificationCreator(notificationCreator);
+
+                MigrationCallback mainThreadReportingMigrationCallback = new MigrationCallback() {
```

Comment:
We make sure that the callbacks occur on the `main thread` again, rather than leaving it to the client.